tests/unit_tests/test_layer_wise_optimizer.py

```diff
@@ -401,3 +402,39 @@ def test_parameter_updates_insufficient_parameters(self):
         This will be insufficient when world size > 2.
         """
         self._run_parameter_update_test(model_class=TinyModel)
+
+    def test_broadcast_vs_allgather(self):
+        """Test LayerWiseDistributedOptimizer allgather code agains broadcast code."""
+        model, optimizer, pg_collection = self.create_model_and_optimizer(model_class=SimpleModel)
+
+        # Create reference model and optimizer using the same function
+        reference_model, reference_optimizer, _ = self.create_model_and_optimizer(
+            model_class=SimpleModel, copy_from=model
+        )
+
+        # Set same gradients on both models
+        for param, ref_param in zip(model.parameters(), reference_model.parameters()):
+            assert torch.equal(param.data, ref_param.data)
+            torch.testing.assert_close(param.data, ref_param.data, rtol=1e-5, atol=1e-5)
```

Comment:
Does it pass with atol=0? It is safer if it can use relative only, i.e. none of the expected values are 0.